target/linux/ar71xx/config-4.4

```diff
@@ -113,6 +113,7 @@ CONFIG_ATH79_MACH_GL_INET=y
 CONFIG_ATH79_MACH_GL_MIFI=y
 CONFIG_ATH79_MACH_GS_MINIBOX_V1=y
 CONFIG_ATH79_MACH_GS_OOLITE=y
+CONFIG_ATH79_MACH_HIVEAP_121=y
```

Comment:
@riptidewave93, it's nothing new, it's just... nobody took care about keeping everything clean and correct here.

The whole ar71xx kernel configuration is just _messy_... as you can see, `generic` subtarget kernel config is almost empty (in fact, it will be empty if you refresh it as the target kernel config already contains that single line from `generic` one) and we use target config for it (yes, ar71xx target kernel config == ar71xx generic subtarget kernel config... and to be honest, I understand why). Then, we have `nand` and `mikrotik` subtargets which are different and should include only related differences in configuration (like enabling boards only related with nand/mikrotik subtargets) but... back to begin - it's messy so instead of keeping `nand` only boards selected in `nand` subtarget kernel config, we have them enabled across all subtargets here (enabled in target config, which results in nand boards code included in... all 3 subtarget kernels: generic, nand and mikrotik).

For now, just make it the correct way (disable in target config, enable in nand subtarget config) please. I have already talked with @nbd168 about possible solutions for this config craziness. One of them is to get rid off all boards Kconfig entries and introduce some kind of group configs for subtargets. With this approach we should end up with clean subtargets configs and make everything less confusing for users.

I just haven't got time to cleanup all of this, too busy this month.

Cheers,
Piotr